packages/graphql/lib/src/links/websocket_link/websocket_client.dart

```diff
@@ -422,18 +458,34 @@ class SocketClient {
                   parse(message.toJson()),
                 ));
 
+        dataErrorComplete
+            .where((message) => message is SubscriptionNext)
+            .cast<SubscriptionNext>()
+            .listen((message) => response.add(
+                  parse(message.toJson()),
+                ));
+
         dataErrorComplete
             .where((message) => message is SubscriptionError)
             .cast<SubscriptionError>()
             .listen((message) => response.addError(message));
 
         if (!_subscriptionInitializers[id]!.hasBeenTriggered) {
-          _write(
-            StartOperation(
-              id,
-              serialize(payload),
-            ),
-          );
+          if (protocol == SocketSubProtocol.graphqlTransportWs) {
+            _write(
+              SubscribeOperation(
+                id,
+                serialize(payload),
+              ),
+            );
+          } else {
+            _write(
+              StartOperation(
+                id,
+                serialize(payload),
+              ),
+            );
+          }
```

Comment:
```suggestion
GraphQLSocketMessage operation =  StartOperation(
            id,
            serialize(payload),
          );
          if (protocol == SocketSubProtocol.graphqlTransportWs) {
            operation = SubscribeOperation(
                id,
                serialize(payload),
              );
          }
```

Reply:
Fixed in https://github.com/maximilianmaihoefner/graphql-flutter/tree/69e46832999656979bf2c6829efc972087c01dff